core/src/main/scala/akka/persistence/r2dbc/query/scaladsl/R2dbcReadJournal.scala

```diff
@@ -197,6 +200,66 @@ final class R2dbcReadJournal(system: ExtendedActorSystem, config: Config, cfgPat
       dbSource
   }
 
+  def eventsBySlicesFromCompactionEvents[Event](
+      entityType: String,
+      minSlice: Int,
+      maxSlice: Int,
+      offset: Offset): Source[EventEnvelope[Event], NotUsed] = {
+    // FIXME DRY with above eventsBySlices
+
+    Source
+      .futureSource {
+        val timestampOffset = toTimestampOffset(offset)
+        queryDao
+          .compactionOffsets(entityType, minSlice, maxSlice, timestampOffset.timestamp)
```

Comment:
Thinking about compaction event vs entity snapshot once more.

Instead of emitting compaction events from the entity, could we load all entity snapshots here and emit them as events (using same seqNr as corresponding event). Would require a transformation function as parameter here that would convert the snapshot to the event type.

Then starting the eventsBySlices query and filtering out earlier events in same way as with the compactionOffsets Map.